schema_validation/validate_schema.go

```diff
@@ -50,21 +51,26 @@ type SchemaValidator interface {
 var instanceLocationRegex = regexp.MustCompile(`^/(\d+)`)
 
 type schemaValidator struct {
-	logger *slog.Logger
-	lock   sync.Mutex
+	options *config.ValidationOptions
+	logger  *slog.Logger
+	lock    sync.Mutex
 }
 
 // NewSchemaValidatorWithLogger will create a new SchemaValidator instance, ready to accept schemas and payloads to validate.
-func NewSchemaValidatorWithLogger(logger *slog.Logger) SchemaValidator {
-	return &schemaValidator{logger: logger, lock: sync.Mutex{}}
+func NewSchemaValidatorWithLogger(logger *slog.Logger, opts ...config.Option) SchemaValidator {
+
+	options := config.NewValidationOptions(opts...)
+
+	return &schemaValidator{options: options, logger: logger, lock: sync.Mutex{}}
+
 }
 
 // NewSchemaValidator will create a new SchemaValidator instance, ready to accept schemas and payloads to validate.
-func NewSchemaValidator() SchemaValidator {
+func NewSchemaValidator(opts ...config.Option) SchemaValidator {
 	logger := slog.New(slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
 		Level: slog.LevelError,
 	}))
-	return NewSchemaValidatorWithLogger(logger)
+	return NewSchemaValidatorWithLogger(logger, opts...)
 }
```

Comment:
NewSchemaValidatorWithLogger was there before this PR, OK

But for me, you could consider to deprecate it and add a WithLogger Option

It doesn't have to be done in this PR, of course.

But I think it would be a huge improvement

But usually, the way to do that is to have a WithDefaultLogger(), that create the logger then it calls WithLogger(logger)

The WithDefaultLogger() is added in the options []Options slice as first argument.

Another way is to check is to do the opposite and appended on the options slice, but the WithDefaultLogger check if the logger is already defined and do nothing if present

Reply:
That would make sense ... 

Thanks again for taking the time to poke through this, I guess the appetite for adopting the fluent style more broadly would rest with the owners.